File: java/03-bootstrap/src/main/java/com/fResult/bootstrap/BootstrapApplication.java
package com.fResult.bootstrap;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;

// @Configuration
// @ComponentScan
// @EnableTransactionManagement
// @Import(DataSourceConfiguration.class)
@SpringBootApplication
public class BootstrapApplication {
  public static void main(String... args) {
    System.setProperty("spring.profiles.active", "prod");
    SpringApplication.run(BootstrapApplication.class, args);
  }

  /* NO-SONAR java:S125
   * public static void main(String[] args) {
   *   final var context = SpringUtils.run(BootstrapApplication.class, "prod");
   *
   *   final var customerService = context.getBean(CustomerService.class);
   *   Demo.workWithCustomerService(BootstrapApplication.class, customerService);
   * }
   *
   * @Bean
   * public PlatformTransactionManager transactionManager(DataSource dataSource) {
   *   return new DataSourceTransactionManager(dataSource);
   * }
   *
   * @Bean
   * public TransactionTemplate transactionTemplate(PlatformTransactionManager transactionManager) {
   *   return new TransactionTemplate(transactionManager);
   * }
   */
}

File: kotlin/03-bootstrap/src/main/kotlin/com/fResult/bootstrap/BootstrapApplication.kt
package com.fResult.bootstrap

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication

// @Configuration
// @ComponentScan
// @Import(DataSourceConfiguration::class)
@SpringBootApplication
class BootstrapApplication {
  /* NO-SONAR
   * @Bean
   * fun transactionManager(dataSource: DataSource): PlatformTransactionManager = DataSourceTransactionManager(dataSource)
   *
   * @Bean
   * fun transactionTemplate(transactionManager: PlatformTransactionManager) = TransactionTemplate(transactionManager)
   */
}

fun main(args: Array<String>) {
  System.setProperty("spring.profiles.active", "prod")
  runApplication<BootstrapApplication>(*args)

  /* NO-SONAR
   * val context = SpringUtils.run(BootstrapApplication::class, "prod")
   * val customerService = context.getBean(CustomerService::class.java)
   *
   * Demo.workWithCustomerService(BootstrapApplication::class, customerService)
   */
}
